docs/large_scale.md

```diff
@@ -0,0 +1,136 @@
+human data set:
+
+```{figure} _static/ancestor_grouping.png
+:width: 80%
+```
+
+There are five tsinfer API methods that can be used to parallelise ancestor
+matching. 
+
+Initially {meth}`match_ancestors_batch_init` should be called to 
+set up the batch matching and to determine the groupings of ancestors.
+This method writes a file `metadata.json` to the `work_dir` that contains
+a JSON encoded dictionary with configuration for later steps, and a key
+`ancestor_grouping` which is a list of dictionaries, each containing the
+list of ancestors in that group (key:`ancestors`) and a proposed partioning of
+those ancestors into sets that can be matched in parallel (key:`partitions`).
+The dictionary is also returned by the method.
+The partitioning is controlled by the `min_work_per_job` and `max_num_partitions`
+arguments. Ancestors are placed in a partition until the sum of their lengths exceeds
+`min_work_per_job`, when a new partition is started. However, the number of partitions
+is not allowed to exceed `max_num_partitions`. It is suggested to set `max_num_partitions`
+to around 3-4x the number of worker nodes available, and `min_work_per_job` to around
+2,000,000 for a typical human data set.
+
+Each group is then matched in turn, either by calling {meth}`match_ancestors_batch_groups`
+to match without partitioning, or by calling {meth}`match_ancestors_batch_group_partition`
+many times in parallel followed by a single call to {meth}`match_ancestors_batch_group_finalise`.
+Each call to {meth}`match_ancestors_batch_groups` or {meth}`match_ancestors_batch_group_finalise`
+outputs the tree sequence to `work_dir`, which is then used by the next group. The length of
+the `ancestor_grouping` in the metadata dictionary determines the group numbers that these methods
+will need to be called for, and the length of the `partitions` list in each group determines
+the number of calls to {meth}`match_ancestors_batch_group_partition` that are needed (if any).
+
+{meth}`match_ancestors_batch_groups` matches groups, without partitioning, from
+`group_index_start` (inclusively) to `group_index_end` (exclusively). Combining
+many groups into one call reduces the overhead from job submission and start
+up times, but note on job failure the process can only be resumed from the
+last `group_index_end`.
+
+To match a single group in parallel, call {meth}`match_ancestors_batch_group_partition`
+once for each partition listed in the `ancestor_grouping[group_index]['partitions']` list,
+incrementing `partition_index`. This will match the ancestors, placing the match data in
+the `working_dir`. Once all are complete a single call to
+{meth}`match_ancestors_batch_group_finalise` will then insert the matches and
+output the tree sequence to `work_dir`.
+
+At anypoint the process can be resumed from the last successfully completed call to 
+{meth}`match_ancestors_batch_groups`. As the tree sequences in `work_dir` checkpoint the
+progress.
```

Comment:
I'm not sure I understand / can parse this last sentence

Reply:
Fixed in 6c02df1 hopefully makes more sense now.